jgrapht-core/src/main/java/org/jgrapht/alg/treedecomposition/ChordalityNiceDecomposition.java

```diff
@@ -0,0 +1,163 @@
+package org.jgrapht.alg.treedecomposition;
+
+import java.util.*;
+
+import org.jgrapht.*;
+import org.jgrapht.alg.cycle.*;
+
+/**
+ * A builder for a nice decomposition for chordal graphs. See {@link NiceDecompositionBuilder} for
+ * an explanation of nice decomposition.
+ * 
+ * @author Ira Justus Fesefeldt (PhoenixIra)
+ *
+ * @param <V> the vertex type of the graph
+ * @param <E> the edge type of the graph
+ */
+public class ChordalityNiceDecomposition<V, E>
+    extends
+    NiceDecompositionBuilder<V>
```

Comment:
when the builder builds the decomposition like in builder design pattern, then it is somehow confusing to extend this builder class.

Reply:
I also do not really see this point here. The class ChordalityNiceDecomposition is in some sense also a builder (possible way more a builder than NiceDecompositionBuilder), since you get a decomposition after specifing your graph. In this sense, the NiceDecompositionBuilder could be classified as the abstract Builder, which the ChordalityNiceDecomposition could be clarified as the concrete Buider. And the dictionary is missing (according to the pattern). 

But since I agreed on renaming it, I think it is more convienient to stay with this structure, since having an associative relation between ChordalityNiceDecomposition and NiceDecompositionBuilder makes the code way less readable.